lib/pbench/server/api/resources/query_apis/__init__.py

```diff
@@ -518,6 +715,69 @@ def _call(self, method: Callable, json_data: Dict[AnyStr, Any]):
             )
             abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")
 
+    def _check_authorization(self, user: str, access: str):
+        """
+        Check whether an API call is able to access data, based on the API's
+        authorization header, the requested user, the requested access
+        policy, and the API's role.
+
+        If "user" is None, then the request is unauthenticated. READ operations
+        may be allowed, UPDATE and DELETE operations will not be allowed.
+
+        If "access" is None, READ will assume we're looking for access only to
+        public datasets.
+
+        for API_OPERATION.READ:
+
+            Any call, with or without an authenticated user token, can access
+            public data.
+
+            Any authenticated user can access their own private data.
+
+            Any authenticated ADMIN user can access any private data.
+
+        for API_OPERATION.UPDATE:
+
+            An authenticated user is required.
+
+            Any authenticated user can update their own data.
+
+            Any authenticated ADMIN user can update any data.
+
+        Args:
+            user: The user parameter to the API, or None
+            access: The access parameter to the API, or None
+
+        Raises:
+            UnauthorizedAccess The user isn't authorized for the requested
+                access.
+        """
+        authorized_user: User = Auth.token_auth.current_user()
```

Comment:
The logic seems compact enough, but in the case of a `READ` operation for `access == public`, can't we just check for that and exit early without causing a database lookup?  Seems like that will be the most often used pattern.

It might also mean you can simplify the logic a bit below.